tests/unit/steps/test_steps_plan.py

```diff
@@ -743,19 +743,122 @@ def test_generate_common_plan(cli_backup, cli_args, model):
             ),
         )
     )
-
-    common_plan = cou_plan.generate_common_plan(target, mock_analysis_result, cli_args)
-
     if cli_backup:
-        expected_plan.add_step(
+        expected_steps.append(
             PreUpgradeStep(
                 description="Backup mysql databases",
                 parallel=False,
                 coro=backup(model),
             )
         )
 
-    assert common_plan == expected_plan
+    pre_upgrade_steps = cou_plan._get_pre_upgrade_steps(mock_analysis_result, cli_args)
+
+    assert pre_upgrade_steps == expected_steps
+
+
+@pytest.mark.parametrize("upgrade_group", ["control-plane", "hypervisor"])
```

Comment:
Please use the global variables instead of those control-plane and hypervisors strings